enclave/src/rpc/mocks/rpc_gateway_mock.rs

```diff
@@ -16,15 +16,60 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
+pub extern crate alloc;
+use alloc::string::String;
+
 use crate::polkadex_balance_storage::Balances;
 use crate::polkadex_gateway::GatewayError;
 use crate::rpc::polkadex_rpc_gateway::RpcGateway;
+use polkadex_sgx_primitives::types::{Order, OrderUUID};
 use polkadex_sgx_primitives::{AccountId, AssetId};
 use sgx_types::{sgx_status_t, SgxResult};
+use substratee_stf::{TrustedCall, TrustedOperation};
 
+/// Mock implementation to be used in unit testing
 pub struct RpcGatewayMock {
     pub do_authorize: bool,
     pub balance_to_return: Option<Balances>,
+    pub order_uuid: Option<OrderUUID>,
+}
+
+/// constructors
+impl RpcGatewayMock {
+    fn default() -> Self {
+        RpcGatewayMock {
+            do_authorize: false,
+            balance_to_return: None,
+            order_uuid: None,
+        }
+    }
+
+    pub fn mock_balances(balances: Option<Balances>, do_authorize: bool) -> Self {
+        let mut get_balances_mock = RpcGatewayMock::default();
+        get_balances_mock.balance_to_return = balances;
+        get_balances_mock.do_authorize = do_authorize;
+        get_balances_mock
+    }
+
+    pub fn mock_place_order(order_uuid: Option<OrderUUID>, do_authorize: bool) -> Self {
+        let mut get_place_order_mock = RpcGatewayMock::default();
+        get_place_order_mock.order_uuid = order_uuid;
+        get_place_order_mock.do_authorize = do_authorize;
+        get_place_order_mock
+    }
+
+    pub fn mock_cancel_order(order_uuid: Option<OrderUUID>, do_authorize: bool) -> Self {
+        let mut get_place_order_mock = RpcGatewayMock::default();
+        get_place_order_mock.order_uuid = order_uuid;
+        get_place_order_mock.do_authorize = do_authorize;
+        get_place_order_mock
+    }
+
+    pub fn mock_withdraw(do_authorize: bool) -> Self {
+        let mut withdraw_mock = RpcGatewayMock::default();
+        withdraw_mock.do_authorize = do_authorize;
+        withdraw_mock
+    }
```

Comment:
Convenience constructors for the gateway mock. If we introduce a mocking framework at some point, it would make this code obsolete.